### PR TITLE
ENH: created functions for simplification

### PR DESCRIPTION
**Pull request by che85**
I don't know if it's useful but I wanted to try

### CODE REVIEW

**Comment by fedorov** [comment]
yes, was thinking about this too, but didn't want to spend the time

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -30,49 +30,49 @@ foreach(json_common_schema_file ${json_common_schema_files})
...
-  add_test(NAME ajv_${json_name}
-           COMMAND ajv compile -s ${json_schema_file} ${common_schema_references} ${AJV_OPTIONS})
+  createSchemaValidationTest(${json_schema_file})
 endforeach()
 
-# validation of each specific (sr, seg, pm) example file including specific schema and referenced common schemata
+# function to simpify creation of example validation tests
+function(createExampleDataTest example_name schema)
+  set(json_example_file ${CMAKE_SOURCE_DIR}/doc/${example_name}-example.json)
+  add_test(NAME ajv_seg-${example_name}
```
Comment (fedorov) [inline doc/CMakeLists.txt]: `seg` should be removed from the test name, as it is object specific. maybe replace with `example`?

Comment (che85) [inline doc/CMakeLists.txt]: Oh yeah makes sense...

File: doc/CMakeLists.txt [hunk 2]
```diff
@@ -30,49 +30,49 @@ foreach(json_common_schema_file ${json_common_schema_files})
   list(APPEND common_schema_references -r ${json_common_schema_file})
 endforeach()
 
+function(createSchemaValidationTest schema_name)
+  get_filename_component(json_name ${schema_name} NAME_WE)
+  add_test(NAME ajv_${json_name}
```
Comment (fedorov) [inline doc/CMakeLists.txt]: maybe also add here `schema` in the name, for consistency

**Comment by fedorov** [comment]
Thanks!

**Pull request merged**